core/db/migrate/20230607100048_create_spree_permission_sets.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class CreateSpreePermissionSets < ActiveRecord::Migration[7.0]
+  def change
+    create_table :spree_permission_sets do |t|
+      t.string :name
+      t.string :set
```

Comment:
I still fail to understand why we need both `name` and `set`. Don't they represent the same information? :thinking:

Reply:
While going through and writing the scope tests this I've noticed a flaw related to this question:
The scope searches by the `name`, meaning that the permission must end with "Display" or "Management" on its `name` attribute in order to be shown by the scope.

I've changed the scope to search by the `set`. This field represents the class name with its namespaces. Which is the key.

The `name` should be something for labeling purposes only, especially for custom permissions. 
Eg.: a user creates a custom permission named "Read/Edit-only custom data" as label on a permission that only allows editing and not managing.

What do you think?

Reply:
I'm also changing `set` to something else. It's a reserved word for many databases, and escaping it doesn't feel right.

Reply:
Update:

I'm changing the `set` attribute to `name` and adding a new attribute named `group` (or label...? WDYT?).

1. `set` is a reserved word for some databases;
2. `name` will now take place as `set` was before; (eg.: `name: "Spree::PermissionSets::UserDisplay"`)
3. `group` will be used for grouping the permissions; (eg.: `Spree::PermissionSet.where(group: "Users")` will return all permissions related to `Users`);
4.  when presenting the permissions for the user when creating a role, we'll displaying the groups with their permissions such as Edit, View or Custom.

![Image 31-08-23 at 04 08](https://github.com/solidusio/solidus/assets/31260493/ff1eff6e-c522-42a9-8fe9-de3786382dee)

Reply:
Thanks, it makes sense